### PR TITLE
feat: Support Nano 33 IoT, MKR WIFI 1010, XIAO, Wio Terminal

### PR DESCRIPTION
**Pull request by kakopappa**
Hi Markus,

1. Support 4 new boards
 - Arduino Nano 33 IoT, MKR WIFI 1010
 - Seeeduino XIAO, Seeeduino Wio Terminal

2. Workflows for Arduino Lint, WiFiNINA Examples, SeedStudio Examples, UNO R4 WiFi Examples 

3. #if defined(SSL_BARESSL) added for function signatures that have BearSSL:: namespace.

4. SSL Support for UNOWIFIR4

### CODE REVIEW

**Comment by kakopappa** [comment]
Hi @Links2004 

What do you think? do I need to do anything

File: src/SocketIOclient.h [hunk 1]
```diff
@@ -55,9 +55,11 @@ class SocketIOclient : protected WebSocketsClient {
     void beginSSL(String host, uint16_t port, String url = "/socket.io/?EIO=3", String protocol = "arduino", uint32_t pingInterval = 60 * 1000, uint32_t pongTimeout = 90 * 1000, uint8_t disconnectTimeoutCount = 5);
 #ifndef SSL_AXTLS
     void beginSSLWithCA(const char * host, uint16_t port, const char * url = "/socket.io/?EIO=3", const char * CA_cert = NULL, const char * protocol = "arduino", uint32_t pingInterval = 60 * 1000, uint32_t pongTimeout = 90 * 1000, uint8_t disconnectTimeoutCount = 5);
-    void beginSSLWithCA(const char * host, uint16_t port, const char * url = "/socket.io/?EIO=3", BearSSL::X509List * CA_cert = NULL, const char * protocol = "arduino", uint32_t pingInterval = 60 * 1000, uint32_t pongTimeout = 90 * 1000, uint8_t disconnectTimeoutCount = 5);
     void setSSLClientCertKey(const char * clientCert = NULL, const char * clientPrivateKey = NULL);
```
Comment (Links2004) [inline src/SocketIOclient.h:58]: `WebSocketsClient::setSSLClientCertKey` code in ifdef but not in header?

File: src/WebSockets.h [hunk 2]
```diff
@@ -248,9 +257,31 @@
 #elif(WEBSOCKETS_NETWORK_TYPE == NETWORK_UNOWIFIR4)
 
 #include <WiFiS3.h>
+#define WEBSOCKETS_NETWORK_CLASS WiFiClient
+#define WEBSOCKETS_NETWORK_SERVER_CLASS WiFiServer
+
+#elif(WEBSOCKETS_NETWORK_TYPE == NETWORK_WIFI_NINA)
```
Comment (Links2004) [inline src/WebSockets.h:263]: WEBSOCKETS_NETWORK_TYPE == NETWORK_WIFI_NINA is defined below already

File: src/WebSocketsClient.cpp [hunk 3]
```diff
@@ -242,7 +244,12 @@ void WebSocketsClient::loop(void) {
         if(_client.isSSL) {
             DEBUG_WEBSOCKETS("[WS-Client] connect wss...\n");
             if(_client.ssl) {
-                delete _client.ssl;
+                #if defined(WIO_TERMINAL) || defined(SEEED_XIAO_M0) || defined(ARDUINO_SAMD_MKRWIFI1010) || defined(ARDUINO_SAMD_NANO_33_IOT)
```
Comment (Links2004) [inline src/WebSocketsClient.cpp:247]: better use the `WEBSOCKETS_NETWORK_TYPE == NETWORK_SAMD_SEED || WEBSOCKETS_NETWORK_TYPE == NETWORK_WIFI_NINA`

File: src/WebSocketsClient.cpp [hunk 4]
```diff
@@ -283,7 +294,11 @@ void WebSocketsClient::loop(void) {
         } else {
             DEBUG_WEBSOCKETS("[WS-Client] connect ws...\n");
             if(_client.tcp) {
+                #if defined(WIO_TERMINAL) || defined(SEEED_XIAO_M0) || defined(ARDUINO_SAMD_MKRWIFI1010) || defined(ARDUINO_SAMD_NANO_33_IOT)
```
Comment (Links2004) [inline src/WebSocketsClient.cpp:297]: better use the `WEBSOCKETS_NETWORK_TYPE == NETWORK_SAMD_SEED || WEBSOCKETS_NETWORK_TYPE == NETWORK_WIFI_NINA`

File: src/WebSocketsClient.h [hunk 5]
```diff
@@ -129,13 +131,20 @@ class WebSocketsClient : protected WebSockets {
...
+    #if defined(SSL_BARESSL)
     BearSSL::X509List * _CA_cert;
     BearSSL::X509List * _client_cert;
     BearSSL::PrivateKey * _client_key;
+    #endif
 #define SSL_FINGERPRINT_IS_SET (_fingerprint != NULL)
 #define SSL_FINGERPRINT_NULL NULL
 #endif
 
+#if defined(WIO_TERMINAL) || defined(SEEED_XIAO_M0) || defined(ARDUINO_SAMD_MKRWIFI1010) || defined(ARDUINO_SAMD_NANO_33_IOT)
```
Comment (Links2004) [inline src/WebSocketsClient.h:143]: WEBSOCKETS_NETWORK_TYPE == NETWORK_SAMD_SEED || WEBSOCKETS_NETWORK_TYPE == NETWORK_WIFI_NINA

**Comment by Links2004** [comment]
had missed the notification

File: src/SocketIOclient.h (hunk 1 above)
Comment (kakopappa) [reply src/SocketIOclient.h:58]: fixed.

**Comment by kakopappa** [comment]
Hi @Links2004. Please take a look again when you get a chance.

**Comment by kakopappa** [comment]
Hi @Links2004 thanks for merging the PR. I was just wondering when we can expect to see the release (didn't mean to rush)

**Pull request merged**